### PR TITLE
Deadlines

### PR DESCRIPTION
**Pull request by sidb28**

### CODE REVIEW

File: src/pages/deadlines/DeadlinesPage.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import { isAssignment } from "../../utils/course";
+
+type Props = DrawerScreenProps<RootDrawerParamList, "Deadlines">;
+
+const DeadlinePage = ({ navigation }: Props) => {
+  const modules = useAppSelector((state) => state.modules);
+  const courses = useAppSelector((state) => state.courses);
+  const deadlines = modules.filter((mod) => isAssignment(mod.type));
+
+  const courseUrlToColors = {}
```
Comment (pranay1208) [inline src/pages/deadlines/DeadlinesPage.tsx]: Add typing for this datastructure. Probably `Record<string, string>` should do

File: src/pages/deadlines/DeadlinesPage.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  for(let i=0; i<courses.length; i++) {
+    courseUrlToColors[courses[i].courseUrl] = courses[i].courseColor
+  }
+
+  const dates = {}
```
Comment (pranay1208) [inline src/pages/deadlines/DeadlinesPage.tsx]: Typing for this too. You might have to define an interface

**Review by pranay1208 — COMMENTED** [review]
Look good to me! Just some minor changes relating to adding typing (TypeScript throws errors for implicit any types)

File: src/pages/deadlines/DeadlinesPage.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { CalendarList } from "react-native-calendars";
+import { useAppSelector } from "../../redux";
+import { isAssignment } from "../../utils/course";
+
+type Props = DrawerScreenProps<RootDrawerParamList, "Deadlines">;
+
+const DeadlinePage = ({ navigation }: Props) => {
+  const modules = useAppSelector((state) => state.modules);
+  const courses = useAppSelector((state) => state.courses);
+  const deadlines = modules.filter((mod) => isAssignment(mod.type));
```
Comment (pranay1208) [inline src/pages/deadlines/DeadlinesPage.tsx:19]: Please add some more filter conditions that:

1) make sure that the due date is non-null, if it is null, then assign it as today
2) make sure that the deadline has not been marked removedFromMoodle or userMarkedDeleted

**Review by pranay1208 — CHANGES_REQUESTED** [review]

File: src/pages/deadlines/DeadlinesPage.tsx (hunk 3 above)
Comment (sidb28) [reply src/pages/deadlines/DeadlinesPage.tsx:19]: If null due dates are assigned to today then the calendar would display dots for all of them on today's date. Better to ignore null values right?

Comment (sidb28) [reply src/pages/deadlines/DeadlinesPage.tsx:19]: @pranay1208 Take a look now

**Review by pranay1208 — APPROVED** [review]

**Pull request merged**